packages/strapi-plugin-content-manager/services/ContentManager.js

```diff
@@ -2,14 +2,17 @@
 
 const _ = require('lodash');
 const uploadFiles = require('../utils/upload-files');
+const getIdAttribute = (model, source) =>
+  _.get(strapi.query(model, source), 'primaryKey', 'id');
 /**
  * A set of functions called "actions" for `ContentManager`
  */
 module.exports = {
   fetch(params, source, populate) {
+    const idAttribute = getIdAttribute(params.model, source);
     return strapi
       .query(params.model, source)
-      .findOne({ id: params.id }, populate);
+      .findOne({ [idAttribute]: params.id }, populate);
```

Comment:
findOne already implements this logic in the background it is not necessary to do this.

Reply:
Thanks. Yes, but `findOne` is compatible with this change. It's not correct to pass here `id` as attribute for PK when you have a table with different PK name. So, this change is reducing further confusion regarding what the field `id` is, when you have no `id` field on a table.

Reply:
The idea is to avoid coupling the content manager with some database specific code.
The content manager must use id as a fixed name to address an entry wheather it's primaryKey is id or not.

If you where calling the database directly your point would stand but the strapi.query layer was made exactly to handle this kind of cases.

Reply:
Got it, thanks for explanation. I've just made those changes